test/solvers/unconstrained.jl

```diff
@@ -0,0 +1,110 @@
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Tridiagonal quadratic" begin
+      n = 100
+      A = spdiagm(0 => 2 * ones(n), -1 => -ones(n-1), -1 => -ones(n-1))
+      nlp = ADNLPModel(x->dot(x, A * x), ones(n))
+
+      stats = with_logger(NullLogger()) do
+        solver(nlp)
+      end
+      @test isapprox(stats.solution, zeros(n), atol=1e-4)
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Rosenbrock" begin
+      nlp = ADNLPModel(x -> (x[1] - 1)^2 + 100 * (x[2] - x[1]^2)^2, [-1.2; 1.0])
+
+      stats = with_logger(NullLogger()) do
+        solver(nlp, max_eval=-1)
+      end
+      @test isapprox(stats.solution, ones(2), atol=1e-6)
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+  end
+
+  @testset "Extended Rosenbrock" begin
+    n = 30
+    nlp = ADNLPModel(x->sum(100 * (x[i+1] - x[i]^2)^2 + (x[i] - 1)^2 for i = 1:n-1), (1:n) ./ (n + 1))
+
+    ng0 = norm(grad(nlp, nlp.meta.x0))
+    stats = with_logger(NullLogger()) do
+      solver(nlp, max_time=600.0, max_eval=-1)
+    end
+    @test isapprox(stats.solution, ones(n), atol=1e-6)
+    @test isapprox(stats.objective, 0.0, atol=1e-6)
+    @test stats.dual_feas < √eps() * (ng0 + 1)
+    @test stats.status == :first_order
+  end
+
+  @testset "Multiprecision" begin
+    for T in (Float16, Float32, Float64, BigFloat)
+      nlp = ADNLPModel(x -> (x[1] - 1)^2 + (x[2] - x[1]^2)^2, T[-1.2; 1.0])
+      ϵ = eps(T)^T(1/4)
+
+      ng0 = norm(grad(nlp, nlp.meta.x0))
+
+      stats = with_logger(NullLogger()) do
+        solver(nlp, max_eval=-1, atol=ϵ, rtol=ϵ)
+      end
+      @test eltype(stats.solution) == T
+      @test stats.objective isa T
+      @test stats.dual_feas isa T
+      @test stats.primal_feas isa T
+      @test isapprox(stats.solution, ones(T, 2), atol=ϵ * ng0 * 10)
+      @test isapprox(stats.objective, zero(T), atol=ϵ * ng0)
+      @test stats.dual_feas < ϵ * ng0 + ϵ
+    end
+  end
+
+  @static if Sys.isunix()
+    @testset "CUTEst" begin
+      problems = CUTEst.select(max_var=2, contype=:unc)
+      stline = statshead([:objective, :dual_feas, :elapsed_time, :iter, :neval_obj,
+                          :neval_grad, :neval_hprod, :status])
+      @info @sprintf("%8s  %5s  %4s  %s\n", "Problem", "n", "type", stline)
+      for p in problems
+        nlp = CUTEstModel(p)
+        stats = with_logger(NullLogger()) do
+          solver(nlp, max_time=3.0)
+        end
+        finalize(nlp)
+
+        ctype = "unc"
+        stline = statsline(stats, [:objective, :dual_feas, :elapsed_time, :iter, :neval_obj,
+                                   :neval_grad, :neval_hprod, :status])
+        @info @sprintf("%8s  %5d  %4s  %s\n", p, nlp.meta.nvar, ctype, stline)
```

Comment:
Is this necessary? What is it testing?

Reply:
Nothing really. It's legacy code. I think we can remove the `CUTEst` problems everywhere.

Reply:
i agree.

Reply:
There is CUTEst test in [trunkls](https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/pull/4/files#diff-c0dce8801f06ebb9221fa66f46546db8R18). Should I remove it, or do you have a suggestion on how to change it?

Reply:
I would replace it with a ADNLSModel.